includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -2813,17 +2813,23 @@ public function update_address_details_for_classic( $payload, $checkout_session_
 		$phone_number = $order->get_shipping_phone();
 		$phone_number = $phone_number ? $phone_number : $order->get_billing_phone();
 
+		$shipping_state = $order->get_shipping_state();
+
 		$payload['addressDetails'] = array(
 			'name'          => rawurlencode( $order->get_shipping_first_name() . ' ' . $order->get_shipping_last_name() ),
 			'addressLine1'  => rawurlencode( $order->get_shipping_address_1() ),
 			'addressLine2'  => rawurlencode( $order->get_shipping_address_2() ),
 			'city'          => rawurlencode( $order->get_shipping_city() ),
-			'stateOrRegion' => rawurlencode( $order->get_shipping_state() ),
+			'stateOrRegion' => rawurlencode( $shipping_state ),
 			'postalCode'    => rawurlencode( $order->get_shipping_postcode() ),
 			'countryCode'   => $order->get_shipping_country( 'edit' ),
 			'phoneNumber'   => rawurlencode( $phone_number ),
 		);
 
+		if ( 'JP' === strtoupper( $payload['addressDetails']['countryCode'] ) && 'JP' === strtoupper( WC_Amazon_Payments_Advanced_API::get_region() ) && isset( self::JP_REGION_CODE_MAP[ $shipping_state ] ) ) {
```

Comment:
Maybe change the `$shipping_state` variable, and using `$order->get_shipping_xxxx()` methods before setting up the payload?

May be simpler, and avoid an unncessary rawurlencode duplication?